src/util/instanceValidation.tsx

```diff
@@ -18,7 +18,7 @@ export const hasNoRootDisk = (
   return missingRoot(values.devices) && !inheritsRoot(values, profiles);
 };
 
-export const isRootDisk = (device: FormDevice): device is FormDiskDevice => {
+export const isRootDisk = (device: FormDevice): boolean => {
```

Comment:
I think this should be reverted. This allowed for an implicit cast, so after `devices.filter(isRootDisk).map((device) => ...` typescript knows that device is of type `FormDiskDevice`.

Reply:
I don't think it's appropriate to have the implicit cast with `isRootDisk` since it's a sub variant of `FormDiskDevice`. I've created the function `isFormDiskDevice` for this purpose. With the old approach, the following condition may result in `device` to get casted with the `never` type.

```
if (device.type !== "disk" || isRootDisk(device)) {
      continue;
    }
```